lib/pool.dart

```diff
@@ -121,6 +121,55 @@ class Pool {
     });
   }
 
+  /// Requests a resource for the combined duration of the [callback] and the
+  /// processing of the Stream it returns.
+  /// (E.g. a database connection is hold until the the results get processed.)
+  Stream/*<T>*/ streamWithResource/*<T>*/(Stream/*<T>*/ callback()) {
```

Comment:
Use the "real" generic method syntax. `streamWithResource<T>` etc

Here and below